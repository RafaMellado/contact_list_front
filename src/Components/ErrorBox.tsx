import { useTranslation } from "react-i18next";

export interface ErrorBoxProps {
  errors: { [key: string]: Array<{ error: string }> };
  field: string;
  params?: object;
}

export function ErrorBox({ errors, field, params }: ErrorBoxProps) {
  const { t } = useTranslation();

  return (
    <>
      {errors[field]?.map((item: { error: string }, index: number) => {
        return (
          <div
            data-testid={`error-box-${field}`}
            className="text-danger mt-2 mb-0"
            key={index}
          >
            <span>
              {t<string>(`errors.${item.error}`, { field, ...params })}
            </span>
          </div>
        );
      })}
    </>
  );
}
