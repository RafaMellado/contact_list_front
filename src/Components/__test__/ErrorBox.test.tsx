import { ErrorBox, ErrorBoxProps } from "../ErrorBox";
import { act, render } from "@testing-library/react";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe(`<${ErrorBox.name} />`, () => {
  const errorBoxProps: ErrorBoxProps = {
    errors: {
      test: [{ error: "test" }],
    },
    field: "test",
  };

  const factoryComponent = (props: ErrorBoxProps = errorBoxProps) =>
    render(<ErrorBox {...props} />);

  test("renders correctly the component", () => {
    const component = factoryComponent();

    act(() => {
      expect(component.asFragment()).toMatchSnapshot();
    });
  });
});
