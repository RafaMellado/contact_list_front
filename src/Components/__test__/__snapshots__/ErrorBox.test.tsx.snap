// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBox /> renders correctly the component 1`] = `
<DocumentFragment>
  <div
    class="text-danger mt-2 mb-0"
    data-testid="error-box-test"
  >
    <span>
      errors.test
    </span>
  </div>
</DocumentFragment>
`;
